app/parse.py

```diff
@@ -15,8 +29,102 @@ class Product:
     num_of_reviews: int
 
 
+def save_to_csv(file_name: str, products: list[Product]) -> None:
+    with open(file_name, "w", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(
+            ["title", "description", "price", "rating", "num_of_reviews"]
+        )
+        for product in products:
+            writer.writerow(
+                [
+                    product.title,
+                    product.description,
+                    product.price,
+                    product.rating,
+                    product.num_of_reviews,
+                ]
+            )
+
+
+class ProductParser:
+    def __init__(self, driver: webdriver.Chrome) -> None:
+        self.driver = driver
+        self.cookies_accepted = False
+
+    def parse_page(self, url: str, file_name: str) -> list[Product]:
+        self.driver.get(url)
+        if not self.cookies_accepted:
+            self.accept_cookies()
+        self.handle_pagination()
+        items = self.driver.find_elements(By.CLASS_NAME, "product-wrapper")
+        products = []
+        with tqdm(total=len(items),
+                  desc="Parsing Products", unit_scale=True) as pbar:
+            for item in items:
+                title = item.find_element(By.CLASS_NAME,
+                                          "title").get_attribute("title")
+                description = item.find_element(
+                    By.CLASS_NAME,
+                    "description"
+                ).text
+                price = float(item.find_element(
+                    By.CLASS_NAME,
+                    "price"
+                ).text[1:])
+                rating = len(item.find_elements(
+                    By.CLASS_NAME,
+                    "ws-icon-star"
+                ))
+                num_of_reviews = int(item.find_element(
+                    By.CLASS_NAME,
+                    "review-count"
+                ).text.split()[0])
+                products.append(Product(title, description,
```

Comment:
if it's a class, then it's better to have a separate method that scrapes specific field